pkg/ami/api.go

```diff
@@ -68,6 +68,11 @@ func Use(ec2api ec2iface.EC2API, ng *api.NodeGroup) error {
 		}
 		if ng.VolumeEncrypted == nil {
 			ng.VolumeEncrypted = output.Images[0].BlockDeviceMappings[0].Ebs.Encrypted
+		} else {
+			// VolumeEncrypted cannot be false if the AMI being used is already encrypted.
+			if api.IsDisabled(ng.VolumeEncrypted) && api.IsEnabled(output.Images[0].BlockDeviceMappings[0].Ebs.Encrypted) {
```

Comment:
Minor nit: `output.Images[0].BlockDeviceMappings[0].Ebs.Encrypted` is used in the corresponding `if` clause, so it'd help to assign it to a variable.

Reply:
@cPu1 do you mean something like this?

```
amiEncrypted := output.Images[0].BlockDeviceMappings[0].Ebs.Encrypted
if ng.VolumeEncrypted == nil {
	ng.VolumeEncrypted = amiEncrypted
} else {
	// VolumeEncrypted cannot be false if the AMI being used is already encrypted.
	if api.IsDisabled(ng.VolumeEncrypted) && api.IsEnabled(amiEncrypted) {
		return fmt.Errorf("%q is an encrypted AMI and volumeEncrypted has been set to false", ng.AMI)
	}
}
```

Reply:
@adamjohnson01 yes.